src/main/java/io/cdap/plugin/jms/common/JMSConfig.java

```diff
@@ -0,0 +1,250 @@
+  @Macro
+  private String jmsUsername;
+
+  @Name(NAME_JMS_PASSWORD)
+  @Description("Password to connect to JMS.")
+  @Macro
+  private String jmsPassword;
+
+  @Name(NAME_PROVIDER_URL)
+  @Description("The URL of the JMS provider. For example, in case of an ActiveMQ Provider, the URL has the format " +
+    "tcp://hostname:61616.")
+  @Macro
+  private String providerUrl;
+
+  @Name(NAME_TYPE)
+  @Description("Queue or Topic.")
+  private String type;
+
+  @Name(NAME_DESTINATION)
+  @Description("Destination Queue/Topic name.")
+  @Macro
+  private String destination;
+
+  @Name(NAME_JNDI_CONTEXT_FACTORY)
+  @Description("Name of the JNDI context factory. For example, in case of an ActiveMQ Provider, the JNDI Context " +
+    "Factory is: org.apache.activemq.jndi.ActiveMQInitialContextFactory.")
+  @Nullable
+  @Macro
+  private String jndiContextFactory;
+
+  @Name(NAME_JNDI_USERNAME)
+  @Description("User name for the JNDI.")
+  @Nullable
+  @Macro
+  private String jndiUsername;
+
+  @Name(NAME_JNDI_PASSWORD)
+  @Description("Password for the JNDI.")
+  @Nullable
+  @Macro
+  private String jndiPassword;
+
+  @Name(NAME_MESSAGE_TYPE)
+  @Description("Supports the following message types: Message, Text, Bytes, Map.")
+  private String messageType;
+
+  @Name(NAME_SCHEMA)
+  @Nullable
+  @Description("Specifies the schema of the records outputted from this plugin.")
+  private String schema;
+
+  public JMSConfig() {
+    super("");
+    this.connectionFactory = "ConnectionFactory";
+    this.type = JMSDestinationType.QUEUE.getName();
+    this.messageType = JMSMessageType.TEXT.getName();
+  }
+
+  @VisibleForTesting
+  public JMSConfig(String referenceName, String connectionFactory, String jmsUsername, String jmsPassword,
+                   String providerUrl, String type, String jndiContextFactory, String jndiUsername,
+                   String jndiPassword, String messageType) {
+    super(referenceName);
+    this.connectionFactory = Strings.isNullOrEmpty(connectionFactory) ? "ConnectionFactory" : connectionFactory;
+    this.jmsUsername = jmsUsername;
+    this.jmsPassword = jmsPassword;
+    this.providerUrl = providerUrl;
+    this.type = Strings.isNullOrEmpty(type) ? JMSDestinationType.QUEUE.getName() : type;
+    this.jndiContextFactory = jndiContextFactory;
+    this.jndiUsername = jndiUsername;
+    this.jndiPassword = jndiPassword;
+    this.messageType = Strings.isNullOrEmpty(messageType) ? JMSMessageType.TEXT.getName() : messageType;
+    this.destination = destination;
+  }
+
+  public String getConnectionFactory() {
+    return connectionFactory;
+  }
+
+  public String getJmsUsername() {
+    return jmsUsername;
+  }
+
+  public String getJmsPassword() {
+    return jmsPassword;
+  }
+
+  public String getProviderUrl() {
+    return providerUrl;
+  }
+
+  public String getType() {
+    return type;
+  }
+
+  public String getJndiContextFactory() {
+    return jndiContextFactory;
+  }
+
+  public String getJndiUsername() {
```

Comment:
should have`@Nullable` annotation for all the fields that can be null

Reply:
Also please make sure callers are handling null values without throwing NPE

Reply:
Done.